### ISSUE TITLE
ibft2 ignores --min-gas-price=0

### ISSUE THREAD

**Issue by danillosl** [hyperledger/besu#3763]
### Description
I'm running a 4 nodes ibft2 from the besu website but apparently the network is ignoring the `--min-gas-price=0`, even though the result of `eth_gasPrice` is `{"jsonrpc" : "2.0", "id" : 53, "result" : "0x0"}` when i try to deploy a contract with truffle i get:

```
> truffle migrate --reset --network besu


Compiling your contracts...
===========================
> Everything is up to date, there is nothing to compile.


Starting migrations...
======================
> Network name:    'besu'
> Network id:      1337
> Block gas limit: 9007199254740991 (0x1fffffffffffff)


1_initial_migration.js
======================

   Replacing 'Migrations'
   ----------------------
   > transaction hash:    0x21e0a743a9cc684d9a6b21ad3b9c4d72e83c9e672040f1b1b99727abf1b0e4cb
   > Blocks: 0            Seconds: 0
   > contract address:    0x42699A7612A82f1d9C36148af9C77354759b210b
   > block number:        10
   > block timestamp:     1650988775
   > account:             0xFE3B557E8Fb62b89F4916B721be55cEb828dBd73
   > balance:             199.999438700537858346
   > gas used:            203143 (0x31987)
   > gas price:           2.763075578 gwei
   > value sent:          0 ETH
   > total cost:          0.000561299462141654 ETH

   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:     0.000561299462141654 ETH


2_deploy_contracts.js
=====================

   Replacing 'contractA'
   --------------------------
   > transaction hash:    0x7ae277c256cb64f8e3487868cdc2f9884d3955ee7c3fe5e47193db08d456bb8e
   > Blocks: 0            Seconds: 0
   > contract address:    0x9a3DBCa554e9f6b9257aAa24010DA8377C57c17e
   > block number:        12
   > block timestamp:     1650988779
   > account:             0xFE3B557E8Fb62b89F4916B721be55cEb828dBd73
   > balance:             199.98087195961082743
   > gas used:            6826558 (0x682a3e)
   > gas price:           2.70141724 gwei
   > value sent:          0 ETH
   > total cost:          0.01844138147105992 ETH


   Replacing 'contractB'
   -------------------------------
   > transaction hash:    0x930918fb20d3ceadeb67db8c9314772c5ee6847cfbc60fede610ce645898c009
   > Blocks: 0            Seconds: 0
   > contract address:    0x9B8397f1B0FEcD3a1a40CdD5E8221Fa461898517
   > block number:        13
   > block timestamp:     1650988781
   > account:             0xFE3B557E8Fb62b89F4916B721be55cEb828dBd73
   > balance:             199.97611977347131766
   > gas used:            1775695 (0x1b184f)
   > gas price:           2.676240086 gwei
   > value sent:          0 ETH
   > total cost:          0.00475218613950977 ETH

   > Saving migration to chain.
   > Saving artifacts
   -------------------------------------
   > Total cost:     0.02319356761056969 ETH

Summary
=======
> Total deployments:   3
> Final cost:          0.023754867072711344 ETH
```

And if a try to deploy with an account that has no funds :

```
Compiling your contracts...
===========================
> Everything is up to date, there is nothing to compile.


Starting migrations...
======================
> Network name:    'besu'
> Network id:      1337
> Block gas limit: 9007199254740991 (0x1fffffffffffff)


1_initial_migration.js
======================

   Replacing 'Migrations'
   ----------------------
 *** Deployment Failed ***

"Migrations" -- Upfront cost exceeds account balance.
```

The deploy log shows that the gas price is around 2.676240086 gwei even tough all nodes are running with --min-gas-price set to zero.

Here is all the files needed to recreate the network [ibft2-example.zip](https://github.com/hyperledger/besu/files/8564984/ibft2-example.zip).

### Acceptance Criteria
Gas price be set to zero.

### Steps to Reproduce
1. Download the ibft2 network example at [ibft2-example.zip](https://github.com/hyperledger/besu/files/8564984/ibft2-example.zip).
2. run each node with the run.sh script inside each node folder.
3. try to deploy a contract.

**Expected behavior:**
Gas price be set to zero.

**Actual behavior:** [What actually happens]
 gas price is around 2.676240086 gwei

### Versions
* Truffle version: `Truffle v5.5.11 (core: 5.5.11)`
* Software version: `besu/v22.1.3/linux-x86_64/openjdk-java-11`
* Java version: `openjdk version "11.0.14.1" 2022-02-08`
* OS Name & Version: `"Linux Mint"`,`"20.2 (Uma)"` 
* Kernel Version: `5.4.0-109-generic`

**Comment by megatronweb3** [comment]
Free gas n/w seem to work for me, all the nodes get stuck with the message - "No sync target, waiting for peers"

**Comment by siladu** [comment]
Hi @danillosl I believe this PR should fix this issue: https://github.com/hyperledger/besu/pull/4003

It's available on the main branch and hopefully it'll make it into the next release. 
In addition to the code in the PR, you must also include `"baseFeePerGas": "0x0"` at the top level of the genesis file.
We will update the documentation to reflect this.

Please reopen or create a new issue if the problem persists even with this code.

**Comment by siladu** [comment]
The fix has been released: https://github.com/hyperledger/besu/releases/tag/22.7.0-RC1
